src/Effect/Logger.hs

```diff
@@ -134,6 +141,33 @@ withWriterLogger sev = runLogger ctx
         , logCtxWrite = tell . pure @f
         }
 
+-- | Like `withWriterLogger`, except it works even when the logged action forks
+-- threads.
+withTVarLogger ::
```

Comment:
[nit]: I would prefer a more generic name.  `withConcurrentLogger` is one idea.

Reply:
That was my original name, but I was concerned that future consumers would see the symbol name and think to themselves that they should be using the concurrent logger to do _any_ concurrent logging, when in fact it's purely a testing utility.

Reply:
If it's purely a testing utility, can we remove it from this module and put it somewhere in `test/`?

Reply:
Well... it's purely a _programmatic usage_ utility. I can't put it into `test/` because I need it exported into Hubble.

Reply:
As a carrier, I like when the names are pretty explicit.  That way we can have other implementations that may also be concurrent without naming conflict.

The function comment could maybe use some more explanation of how it's using a TVar so that readers understand why they would want to use a TVar logger and what the limitations are.  For example, I'm not clear what the interleaving semantics are just from the naming.